src/legacy/core_plugins/metrics/public/components/panel_config/gauge.js

```diff
@@ -117,30 +123,34 @@ class GaugePanelConfigUi extends Component {
           <EuiPanel>
             <EuiTitle size="s">
               <span>
-                <FormattedMessage
-                  id="tsvb.gauge.optionsTab.dataLabel"
-                  defaultMessage="Data"
-                />
+                <FormattedMessage id="tsvb.gauge.optionsTab.dataLabel" defaultMessage="Data" />
               </span>
             </EuiTitle>
             <EuiSpacer size="m" />
 
-            <IndexPattern
-              fields={this.props.fields}
-              model={this.props.model}
-              onChange={this.props.onChange}
-            />
+            <UIRestrictionsContext.Consumer>
```

Comment:
@Avinar-24  I see you added <UIRestrictionsContext.Consumer> for all IndexPattern components. What if we move Context Consumer into IndexPattern component?

Reply:
@alexwizp Nice catch. But there is another way exists. We may substitute `Context.Consumer` by the `useContext` hook. It looks better and clearer.